assets/model_monitoring/components/src/action_analyzer/action_analyzer_output_actions/run.py

```diff
@@ -4,20 +4,22 @@
 """Entry script for Action Analyzer output actions."""
 
 import argparse
+from action_analyzer.contracts.utils.detector_utils import (
+    write_to_file
+)
 from pyspark.sql.types import (
     StringType,
     BooleanType
 )
 from pyspark.sql import Window
 from pyspark.sql.functions import collect_set, col, udf, max
-from shared_utilities.io_utils import try_read_mltable_in_spark, np_encoder
+from shared_utilities.store_url import StoreUrl
+from shared_utilities.io_utils import try_read_mltable_in_spark
 import os
-import json
 import uuid
 import datetime
 import copy
 from mlflow import MlflowClient
-from shared_utilities.amlfs import amlfs_upload
```

Comment:
are we going to remove the dependency on azureml.fsspec? 

1. are we going to remove the amlfs.py?
2. after that, are we going to remove the conda package azureml-fsspec?

Reply:
1. No, the create manifest and output metrics components will still need to use functions from amlfs.py that are not supported by StoreUrl such as downloading/uploading whole folders. For these components I have to manually set some env vars to support credential-less data save/read. In most other scenarios I am moving our code to use StoreUrl which will handle credential-less datastore support on-demand in a cleaner way than adhoc env variables initialization.
2. I was considering this but due to the cache issues we have seen, I think I'll leave it in place for now. Can reconsider removing that dependency later on if we can depend on cache being updated.

Reply:
I am testing to see which components we can remove fsspec dependency from in case we want to remove in this PR.